File: Data Structures & Algorithms/two-integer-sum/submission-14.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> diffs = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++){
            diffs[nums[i]] = i;
        }
        int[] res = new int[2];
        for(int i = 0; i < nums.Length; i++){
            int diff = target - nums[i];
            if(diffs.ContainsKey(diff) && diffs[diff] != i){
                res[0] = Math.Min(i, diffs[diff]);
                res[1] = Math.Max(i, diffs[diff]);
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-14.java
class Solution {
    public int characterReplacement(String s, int k) {
        Set<Character> set = new HashSet<>();
        for(int i = 0; i < s.length(); i++){
            set.add(s.charAt(i));
        }
        int res = 0;
        for(char c : set){
            int l = 0;
            int count = 0;
            for(int r = 0; r < s.length(); r++){
                if(s.charAt(r) == c){
                    count++;
                }

                while(r - l + 1 - count > k){
                    if(s.charAt(l) == c){
                        count--;
                    }
                    l++;
                }

                res = Math.max(r - l + 1, res);

            }
        }
        return res;
    }
}
